Docs for rejecting compromised passwords on sign-in (#874)
clerk/clerk-docs: docs/security/password-protection.mdx

Also, please bear in mind, that passwords are not a requirement for using Clerk. Applications can be configured to use a passwordless strategy that relies on your users being sent one-time passwords instead.

## Reject compromised passwords
Clerk refers to the National Institute of Standards and Technology (NIST) guidelines to determine its handling of compromised passwords:

When processing requests to establish and change memorized secrets, verifiers SHALL compare the prospective secrets against a list that contains values known to be commonly-used, expected, or compromised. For example, the list MAY include, but is not limited to: [NIST Special Publication 800-63B](https://pages.nist.gov/800-63-3/sp800-63b.html#sec5)

- Passwords obtained from previous breach corpuses.

> Specifically, Clerk contracts with have [i been pwned](https://haveibeenpwned.com/) to compare prospective passwords against its corpus of over 10 billion leaked credentials.

Reviewer: ```suggestion
Specifically, Clerk contracts with [HaveIBeenPwned](https://haveibeenpwned.com/) to compare prospective passwords against its corpus of over 10 billion leaked credentials.
```

> Rejection of compromised passwords is enabled by default on all Clerk apps, for sign-up and password changes.

Reviewer: ```suggestion
Rejection of compromised passwords is enabled by default for sign-up and password changes. It is disabled by default for sign-in - learn more in the [Reject compromised passwords on sign-in] section.
```

> Settings for this feature can be found in the Clerk Dashboard under **User & Authentication > [Email, Phone, Username](https://dashboard.clerk.com/last-active?path=user-authentication/email-phone-username)**, in the **Authentication strategies > Password** section.

Reviewer 2: ```suggestion
You can configure this feature in the Clerk Dashboard under **User & Authentication > [Email, Phone, Username](https://dashboard.clerk.com/last-active?path=user-authentication/email-phone-username)**, in the **Authentication strategies > Password** section.
```

Reviewer: ```suggestion
To configure this feature:

1. Navigate to the [Clerk Dashboard](https://dashboard.clerk.com/last-active?path=user-authentication/email-phone-username).
2. In the navigation sidenav, select **User & Authentication > Email, Phone, Username**.
3. In the **Authentication strategies** section, next to **Password**, select the settings cog icon.
4. You can enable or disable **Reject compromised passwords on sign-up or password change, powered by HaveIBeenPwned**. You can also enable or disable **Reject compromised passwords also on sign-in".
```

> ![Settings for compromised password handling](/public/images/authentication/reject_compromised_passwords.png)

Reviewer 2: ```suggestion
```

In general, we avoid screenshots unless they're required for understanding the information being presented

Author: Removing

### Reject compromised passwords on sign-in

> As can be seen in the above screenshot, Clerk also offers the capability to check for compromised passwords upon sign-in.

Reviewer 2: ```suggestion
You can check for compromised passwords on sign-in.
```

> This is useful for blocking password sign-ins in the case that:

Reviewer 2: ```suggestion
```
No longer needed due to previous edit

> * the password has recently been added to the compromised password database

Reviewer 2: ```suggestion
- The password has recently been added to the compromised password database
```

* the user was able to set a compromised password because protection was off at the time

> * the user was migrated to Clerk along with their existing password digest

Reviewer 2: ```suggestion
- The user was migrated to Clerk along with their existing password digest
```

> If enabled, when the user provides their correct password that nevertheless has been found in online breach data, they will not be signed in, but instead prompted to reset their password.

Reviewer 2: ```suggestion
```
No longer needed due to previous edit (moved this to the start of the section)

> Password reset for compromised passwords utilizes the same flow as for "forgot password", i.e. the user will need to authenticate first via an OTP code on their email or phone and only then they will be able to set a new, more secure, password.

Reviewer 2: ```suggestion
<Callout type="Info">
Password reset for compromised passwords uses the same flow as "forgot password". The user will need to authenticate first via an OTP code on their email or phone and only then they will be able to set a new, more secure, password.
</Callout>
```

> **Disclaimers:**

Reviewer 2: ```suggestion
#### Limitations
```

> * Prior to enabling rejection of compromised passwords on sign-in, please ensure your app has support for the password reset flow, either by utilizing Clerk components or via your [own custom implementation](/docs/custom-flows/forgot-password)

Reviewer 2: ```suggestion
- Before enabling rejection of compromised passwords on sign-in, please ensure your app has support for the password reset flow. You can do so by using Clerk components or [implementing a custom flow](/docs/custom-flows/forgot-password).
```

> * If the user has no way of resetting their password, e.g. if the application configuration allows for a user to be without a communication method, then Clerk will not enforce this restriction for the impacted user

Reviewer 2: ```suggestion
- If the user has no way of resetting their password, e.g. if your application does not require an email, phone number, or other communication method to sign up, Clerk will not reject compromised passwords on sign-in.
```

Reviewer: Remove the use of "e.g." as it might not be understood in other languages/cultures.

```suggestion
- If the user has no way of resetting their password—such as when your application does not require an email, phone number, or other communication method for sign-up—Clerk will not reject compromised passwords on sign-in.
```
